woocommerce/wc-admin
# Try playwright for e2e tests

Partially fixes #6313 

Update the e2e tests to use [Playwright](https://playwright.dev/) and convert tests to Typescript.
Also added models directory with classes used as page objects.

### Screenshots

Short video with the playwright inspector -> https://d.pr/v/gKZnts

<img width="1055" alt="Screen Shot 2021-03-02 at 2 50 14 PM" src="https://user-images.githubusercontent.com/2240960/109699483-03c4a900-7b67-11eb-959a-659bab9d5e63.png">

### Detailed test instructions:

- Load this PR and run `npm install`
- Run `npm run docker:up`
- Run `npm run test:e2e`, it should run all the tests and pass
- Run `npm run test:e2e-dev`, it should show a chromium browser and run the tests (one of the tests might fail the second time around, given we can't reset the DB yet).
- Run `npm run test:e2e tests/e2e/specs/activate-and-setup/setup-wizard.test.ts`, it should just run that test file.
- Try out the Playwright debugger, by adding `await page.pause()` [here](https://github.com/woocommerce/woocommerce-admin/blob/try/6313_playwright_e2e/tests/e2e/specs/activate-and-setup/complete-store-details-section.ts#L36) and running `npm run test:e2e-dev`, it should pause there and show the playwright inspector and allows you to step through the tests -> https://playwright.dev/docs/inspector
